src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,40 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userByEmail = userService.findByEmail(email);
+        if (userByEmail.isPresent() && HashUtil.hashPassword(password, userByEmail.get().getSalt())
+                .equals(userByEmail.get().getPassword())) {
+            return userByEmail.get();
+        }
+        throw new AuthenticationException("Can't login a user with email " + email);
+    }
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
```

Comment:
[checklist](https://mate-academy.github.io/jv-program-common-mistakes/hibernate/add-user-service/add-user-service-checklist): We need to check the email during user registration and make the email a unique field in the database.

lets check if we have user with such login in db first